src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/KubernetesCredentialProvider.java

```diff
@@ -163,6 +171,16 @@ public void stopWatchingForSecrets() {
         if (ACL.SYSTEM.equals(authentication)) {
             ArrayList<C> list = new ArrayList<>();
             for (IdCredentials credential : credentials.values()) {
+                if (itemGroup != Jenkins.getInstanceOrNull()) {
+                    String scope = '/' + itemGroup.getUrl();
+                    Collection<String> validScopes = credentialScopes.get(credential.getId());
+                    LOG.log(Level.FINEST, "getCredentials checking if {0} has scope {1}", new Object[] { credential.getId(), scope });
+                    if (!validScopes.isEmpty() && validScopes.stream().noneMatch(scope::startsWith)) {
```

Comment:
startsWith does not seem like a good idea. imagine you provide an annotation with "team1" and folder "team12" receives the credential. If this resolver should work like the folders plugin (credentials are accessible for subfolders), the path needs either to be tokenized and prefix-matched - or the annotation entry must forced to end with a  trailing `/`.